src/coord/src/catalog.rs

```diff
@@ -2201,23 +2222,35 @@ impl Catalog {
         let SerializedCatalogItem::V1 {
             create_sql,
             eval_env: _,
-            persist_name,
+            table_persist_name,
+            source_persist_details,
         } = serde_json::from_slice(&bytes)?;
-        self.parse_item(id, create_sql, Some(&PlanContext::zero()), persist_name)
+        self.parse_item(
+            id,
+            create_sql,
+            Some(&PlanContext::zero()),
+            table_persist_name,
+            source_persist_details,
+        )
     }
 
+    // Parses the given SQL string into a `CatalogItem`.
+    //
+    // The given `persist_details` are an optional description of the persisted streams that this
+    // source uses, if it is a persisted source.
     fn parse_item(
         &self,
         id: GlobalId,
         create_sql: String,
         pcx: Option<&PlanContext>,
-        persist_name: Option<String>,
+        table_persist_name: Option<String>,
+        source_persist_details: Option<SerializedSourcePersistDetails>,
     ) -> Result<CatalogItem, anyhow::Error> {
         let stmt = sql::parse::parse(&create_sql)?.into_element();
         let plan = sql::plan::plan(pcx, &self.for_system_session(), stmt, &Params::empty())?;
         Ok(match plan {
             Plan::CreateTable(CreateTablePlan { table, .. }) => {
```

Comment:
let's be super defensive here and return an error if we get a Some for source_persist_details and the plan is a CreateTable and similarly if we get a Some for table_persist_name and the plan is a CreateSource

Reply:
:ok_hand: